Updated cppcheck and corrected some errors that it found (#911)
SINTEF/dlite: doc/user_guide/transactions.md

Transactions
============

> Transaction is a concept that comes from SOFT, which allows to easy manage arbitrary long series of immutable (frozen) instances, while ensuring provenance.

Reviewer: Weird grammar

> Transaction is a concept that comes from SOFT, which allows to easy manage arbitrary long series of immutable (frozen) instances, while ensuring provenance.

Reviewer: ```suggestion
The Transaction concept comes from SOFT. It allows to easily manage arbitrary long series of immutable (frozen) instances, while ensuring provenance.
```
